### ISSUE TITLE
Separate subsession identifiers from other identifiers

### ISSUE THREAD

**Issue by marsella** [boltlabs-inc/tss-ecdsa#162]
One use of the `Identifier` type is as a subsession identifier: the unique ID attached to a single instance of a running subsession. The `Identifier` type doesn't clearly document this right now.

- [ ] Update documentation on the `Identifier` type to explain its role in the protocol (global, unique, flags messages and makes proofs unique from each other). Optionally, rename to `SessionId` or `SubsessionId` (I'm not sure if this is necessary for the developer or if improved documentation is enough. You can decide.)
- [ ] Review documentation on fields of type `Identifier` and update if they're not clear about the purpose of the ID. E.g. I think the documentation in `Message` is confusing right now -- it's not an ID for the message contents, it's a session ID for the session that the message belongs to.
- [ ] Revisit the `ParticipantIdentifier` type. Decide whether it should be a wrapper around `Identifier`, or if that's too confusing and it should be its own `u128`.

[See discussion](https://github.com/boltlabs-inc/tss-ecdsa/issues/140#issuecomment-1419848849).

**Comment by mattlebeau-galois** [comment]
@LordQ1 - Do we have more specific acceptance criteria for this?

**Comment by Ra1issa** [comment]
One thing I recommend doing here is turning the `Identifier` struct into a trait that things like `SessionIdentifier` and `SubSessionIdentifier` implement. It could be a cleaner way to abstract away the shared behavior between different identifiers while also specifying what kind of identifier ever method would nee.

My proposed API would look something like this: 

```
pub trait Identifier {
    fn new(value: u128) -> Self;
    fn random<R: RngCore + CryptoRng>(rng: &mut R) -> Self;
}
```

The something like a `SessionIdenfitier` could implement by doing something like: 

```
pub struct SessionIdenfitier {
 pub value: u128,
}

impl Identifier for SessionIdenfitier{
    pub fn new(value: u128) -> Self {
        // do something  with value
        Self {
         value,
       }
    }
    pub fn random<R: RngCore + CryptoRng>(rng: &mut R) -> Self {
        Self {
         value: rng.gen::<u128>(),
       }
    }
}

impl std::fmt::Display for SessionIdenfitier {
    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
        write!(f, "Session Id({})", hex::encode(&self.0.to_be_bytes()[..4]))
    }
}
```
and similarly for `SubSessionIdentifier` and maybe even `ParticipantIdentifier`.

**Comment by marsella** [comment]
Closed by #225.
